src/Codeception/Lib/InnerBrowser.php

```diff
@@ -504,7 +504,7 @@ public function seeLink($text, $url = null)
             $this->fail("No links containing text '$text' were found in page " . $this->_getCurrentUri());
         }
         if ($url) {
-            $crawler = $crawler->filterXPath(sprintf('.//a[contains(@href, %s)]', Crawler::xpathLiteral($url)));
+            $crawler = $crawler->filterXPath(sprintf('.//a[substring(@href, string-length(@href) - string-length(%1$s) + 1)=%1$s]', Crawler::xpathLiteral($url)));
```

Comment:
XPath `ends-with` is only supported in XPath 2.0, and PHP has only XPath 1.0.

See:
1. https://stackoverflow.com/questions/11848780/use-ends-with-in-xslt-v1-0
1. https://secure.php.net/manual/en/class.domxpath.php